services/src/main/java/org/keycloak/protocol/oidc/DefaultTokenExchangeProvider.java

```diff
@@ -311,10 +316,28 @@ protected Response exchangeClientToClient(UserModel targetUser, UserSessionModel
             }
         }
 
-        if (targetClient.isConsentRequired()) {
-            event.detail(Details.REASON, "audience requires consent");
-            event.error(Errors.CONSENT_DENIED);
-            throw new CorsErrorResponseException(cors, OAuthErrorException.INVALID_CLIENT, "Client requires user consent", Response.Status.BAD_REQUEST);
+        String scopesFromSubjectToken = null;
```

Comment:
As mentioned in the other PR, the requested scopes are already available from here https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/protocol/oidc/DefaultTokenExchangeProvider.java#L342.

Reply:
fixed, check my changes please. I move this part of code after calculation of scopes.